### PR TITLE
calc: don't jump to own cursor if not looking

### PR DESCRIPTION
**Pull request by eszkadev**
If we have cell selected on the bottom of the sheet and we scroll up to watch something else so we don't see the cursor: if other user will add row above - we will jump to our own cursor. It's because it changed the address: like A500 to A5001.

Let's don't do that. Solution could be to enter "non following" mode when we scroll outside our own cursor. So we will not jump to our cell if not needed. Then if we click or do any key press - we should cancel that mode and go back to "following own cursor mode".

This can be additionaly marked by something on the user list.

### CODE REVIEW

**Comment by eszkadev** [comment]
hi, @gokaysatir & @Minion3665 could you look at some point - this patch needs other eyes and manual testing :)

Affects:
- following users / editor
- scrolling in calc

**Review by gokaysatir — APPROVED** [review]
Code seems ok and works beautifully for me, thanks :)

**Review by Minion3665 — CHANGES_REQUESTED** [review]
This patch seems to stop me from unfollowing someone by typing (I tested in Calc) ... (there's also a following problem where the unfollowed state still shows in the chip and is called "Following You", but I think this is an earlier regression)

**Review by Minion3665 — APPROVED** [review]
working for me now and the change looks good, thanks!

File: browser/src/layer/tile/CanvasTileLayer.js [hunk 1]
```diff
@@ -2263,7 +2277,11 @@
 
 		var scrollToCursor = this._sheetSwitch.tryRestore(sameAddress, this._selectedPart);
 
-		this._onUpdateCellCursor(scrollToCursor, sameAddress);
+		var isFollowingOwnCursor = parseInt(app.getFollowedViewId()) === parseInt(this._viewId);
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply browser/src/layer/tile/CanvasTileLayer.js:2278]: done

**Pull request merged**